GoogleForCreators/web-stories-wp
# Editor: Implement MVP for Post Locking (Story Locking)

## Context



## Summary

A very basic implementation of post locking in the editor. 

Please note, the code is very messy and needs a lot of tidying up. The idea was just to get this thing working. 


## Relevant Technical Choices

Some of the fields adding to the settings object, maybe loaded from other places. But as the post locking is loaded very early.

No new endpoint was created, I used extended existing endpoints.  

User data is loaded via an embedded REST API request. 

Post lock data can not be preloaded as it caches.

## To-do

Should we implement delete lock on navigation away from story editor?
Should this feature be behind a feature flag?
Should the links be tracked?


## User-facing changes


https://user-images.githubusercontent.com/237508/109031809-f9b12f00-76bc-11eb-8b1a-9152f28559af.mov



<img width="1884" alt="Screenshot 2021-02-23 at 18 38 56" src="https://user-images.githubusercontent.com/237508/108892045-52be8b80-7607-11eb-8201-197ec345963d.png">

<img width="1884" alt="Screenshot 2021-02-24 at 11 53 41" src="https://user-images.githubusercontent.com/237508/108997233-3ff39780-7697-11eb-91b3-140f8dc8f849.png">

<img width="270" alt="Screenshot 2021-02-23 at 18 39 09" src="https://user-images.githubusercontent.com/237508/108892122-6bc73c80-7607-11eb-8c65-42629bced454.png">

## Testing Instructions



### QA






This PR can be tested by following these steps:

1. Login as author, 
2. Create a story. 
3. Publish story
4. In another browser, login as admin. 
5. Edit the same story. 
6. See dialog. 

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #3126
